File: server/models/cart_item.py
import uuid
from sqlalchemy.sql import func
from . import db

class CartItem(db.Model):
    __tablename__ = 'cart_items'
    
    id = db.Column(db.String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    cart_id = db.Column(db.String(36), db.ForeignKey('carts.id', ondelete='CASCADE'), nullable=False)
    product_id = db.Column(db.String(36), db.ForeignKey('products.id', ondelete='CASCADE'), nullable=False)
    quantity = db.Column(db.Integer, nullable=False, default=1)
    created_at = db.Column(db.DateTime, default=func.now(), nullable=False)
    updated_at = db.Column(db.DateTime, default=func.now(), onupdate=func.now(), nullable=False)

    # Relationships
    product = db.relationship('Product', back_populates='cart_items')
    
    def to_dict(self):
        """Convert cart item object to dictionary"""
        return {
            'id': self.id,
            'cart_id': self.cart_id,
            'product_id': self.product_id,
            'quantity': self.quantity,
            'created_at': self.created_at.isoformat() if self.created_at else None,
            'updated_at': self.updated_at.isoformat() if self.updated_at else None,
            'product': self.product.to_dict() if self.product else None
        }

File: server/models/cart_item.ts
import { DataTypes, Model, Optional, Sequelize } from 'sequelize';

interface CartItemAttributes {
  id: string;
  cart_id: string;
  product_id: string;
  quantity: number;
}

export interface CartItemCreationAttributes extends Optional<CartItemAttributes, 'id'> {}

export class CartItem extends Model<CartItemAttributes, CartItemCreationAttributes> implements CartItemAttributes {
  public id!: string;
  public cart_id!: string;
  public product_id!: string;
  public quantity!: number;

  // timestamps
  public readonly created_at!: Date;
  public readonly updated_at!: Date;

  public static associate: (models: any) => void;
}

export function initCartItem(sequelize: Sequelize) {
  CartItem.init(
    {
      id: {
        type: DataTypes.STRING(36), // UUID
        primaryKey: true,
        defaultValue: DataTypes.UUIDV4,
      },
      cart_id: {
        type: DataTypes.STRING(36),
        allowNull: false,
        references: { model: 'carts', key: 'id' },
        onDelete: 'CASCADE',
      },
      product_id: {
        type: DataTypes.STRING(36),
        allowNull: false,
        references: { model: 'products', key: 'id' },
        onDelete: 'CASCADE',
      },
      quantity: { type: DataTypes.INTEGER, allowNull: false },
    },
    {
      sequelize,
      modelName: 'CartItem',
      tableName: 'cart_items',
      timestamps: true, // Let Sequelize manage timestamps
      underscored: true // Use snake_case for all field names
    }
  );

  CartItem.associate = function(models) {
    CartItem.belongsTo(models.Cart, {
      foreignKey: 'cart_id',
      as: 'cart',
    });
    CartItem.belongsTo(models.Product, {
      foreignKey: 'product_id',
      as: 'product',
    });
  };

  return CartItem;
}
